crates/solver/src/driver.rs

```diff
@@ -374,35 +368,41 @@ impl Driver {
         let current_block_during_liquidity_fetch =
             current_block::block_number(&self.block_stream.borrow())?;
 
-        let orders = self.api.get_orders().await.context("get_orders")?;
-        let (before_count, block) = (orders.orders.len(), orders.latest_settlement_block);
-        let orders = self.in_flight_orders.update_and_filter(orders);
-        if before_count != orders.len() {
+        let mut auction = self.api.get_auction().await.context("get_auction")?;
+        let before_count = auction.orders.len();
+        self.in_flight_orders.update_and_filter(&mut auction);
+        if before_count != auction.orders.len() {
             tracing::debug!(
                 "reduced {} orders to {} because in flight at last seen block {}",
                 before_count,
-                orders.len(),
-                block
+                auction.orders.len(),
+                auction.block
             );
         }
-        let orders = orders
+
+        let orders = auction
+            .orders
             .into_iter()
             .map(|order| self.order_converter.normalize_limit_order(order))
             .collect::<Vec<_>>();
         tracing::info!("got {} orders: {:?}", orders.len(), orders);
+
+        let estimated_prices = auction
+            .prices
+            .into_iter()
+            .map(|(token, price)| {
+                (
+                    token,
+                    price.to_big_rational() / BigInt::from(1_000_000_000_000_000_000_u128),
```

Comment:
My understanding of the `estimated_prices` were that they were exchange rates from Ether to the token. Since the prices are denominated in Ether with 18 decimals (i.e. the amount of Ether required to buy `1e18` of the token) we need to divide by `1e18` to get an actual exchange rate (such that `X ETH * estimated_prices[TOK] = Y TOK`).

Reply:
Looks correct:
```rust
        .map(|token| price_estimation::Query {
            // For ranking purposes it doesn't matter how the external price vector is scaled,
            // but native_token is used here anyway for better logging/debugging.
            sell_token: native_token,
            buy_token: token,
            in_amount: native_token_amount_to_estimate_prices_with,
            kind: model::order::OrderKind::Sell,
        })

    // ...

            let price = match estimate.price_in_sell_token_rational(&query) {

    // ...

    // Always include the native token.
    prices.insert(native_token, num::one());
    // And the placeholder for its native counterpart.
    prices.insert(BUY_ETH_ADDRESS, num::one());
```

So it looks like the prices in the `estimated_prices` maps are indeed exchange rates from token to ETH. Since the service returns prices in ETH with 18 decimals, this is computed by: `price / 1e18`.

I will add a small helper function to `auction_preprocessing` with comments to make this more obvious.

Reply:
Added a separate method with tests that hopefully clarify the format used in the `estimated_prices` mapping.